bin/deepstate/main_evm.py

```diff
@@ -0,0 +1,144 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+logging.basicConfig()
+
+import collections
+import sys
+import os
+import md5
+
+try:
+  import manticore.ethereum
+except Exception as e:
+  if "Z3NotFoundError" in repr(type(e)):  
+    print "Manticore requires Z3 to be installed."
+    sys.exit(255)
+  else:
+    raise
+import traceback
+from .common import DeepState
+
+L = logging.getLogger("deepstate.evm")
+L.setLevel(logging.INFO)
+
+def move_tests(workspace, outdir):
+  """Parse and sort testcases"""  
+  tx_files = []
+  for prefix,d,files in list(os.walk(workspace)):
+    for f in files: 
+      if "test" in f and ".tx" in f:
+        tx_files.append(prefix+"/"+ "/".join(d) + f)
+
+  for filename in tx_files:
+      txs = open(filename, "r+").read()
+      test_name = md5.new(txs).hexdigest()
+      if "THROW" in txs:
+          test_name = test_name + ".fail"
+      if "STOP" in txs:
+          test_name = test_name + ".pass" 
+
+      test_file = os.path.join(outdir, test_name)
+      L.info("Saving input to {}".format(test_file))
+      try:
+        with open(test_file, "wb") as f:
+          f.write(txs)
+      except:
+        L.critical("Error saving input to {}".format(test_file))
+
+def compile_contract(contract):
+  """Compile a contract"""
+  m = manticore.ethereum.ManticoreEVM()
+  source_code = open(contract,"r").read()
+  owner_account = m.create_account(balance=1000)
+  contract_account = m.solidity_create_contract(source_code, owner=owner_account, contract_name="TEST")
+  return m, owner_account, contract_account
+ 
+def do_run_test(args, contract, test):
+  """Run an individual test case."""
+  test_name, test_args = test
+  m, owner_account, contract_account = compile_contract(contract)
+  m.verbosity(1)
+  func = contract_account.__getattribute__(test_name)
+  xs = [None]*len(test_args)
+  func(*xs)
+  m.finalize()
+  move_tests(m.workspace, mk_test_dir(args, test))
+
+def run_test(args, contract, test):
+  try:
+    do_run_test(args, contract, test)
+  except:
+    L.error("Uncaught exception: {}\n{}".format(
+        sys.exc_info()[0], traceback.format_exc()))
+
+def find_test_cases(contract):
+  """Iterate over all the methods in the TEST contract and collect the "Test_" ones"""
+  m, owner_account, contract_account = compile_contract(contract) 
+  signatures = m.get_metadata(contract_account.address).signatures
+  test_cases = []
+  for (h, n) in signatures.items():
+    if "Test_" in n:
+      test_name = n.split("(")[0]
+      test_args = n.split("(")[1].split(")")[0].split(",")
+      test_cases.append((test_name,test_args))
+
+  return test_cases
+
+def mk_test_dir(args, test):
```

Comment:
This function's name implies that it creates the test directory, which it does not.

Reply:
Later on you do try to use `os.makedirs(test_dir)`. You should rename this function to: `try_make_test_dir`, and inline the `os.makedirs` into the function.